### PR TITLE
[3267] Fix potential NPE in TreeComponent

### LINKED ISSUE #3267: Potential NPE in TreeComponent
* [x] **I have checked that this bug has not yet been reported by someone else**
* [x] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- latest
* [x] **I have specified my environment** : 
- All

In `TreeComponent#computeChildren`, the list of `semanticChildren` may contain `null` values.
The `semanticChildren` is then used in a loop and each `null` value may lead to NPE.
 
### Expected behavior

Filter `null` values from the `semanticChildren`.

### Actual behavior

No filter on `null` values from the `semanticChildren`.

### PR DESCRIPTION
**Pull request by AxelRICHARD**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/3267

# Pull request template

## General purpose
What is the main goal of this pull request?
- [x] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [x] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

**Review by pcdavid — APPROVED** [review]

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -115,6 +115,7 @@ The implemented fix does not allow to evaluate an AQL expression, but only to re
 - https://github.com/eclipse-sirius/sirius-web/issues/3206[#3206] [diagram] Fix an issue where precondition on tool was not respected during palette creation
 - https://github.com/eclipse-sirius/sirius-web/issues/3257[#3257] [diagram] Fix an issue that caused an infinite loop when selecting elements with shift.
 - https://github.com/eclipse-sirius/sirius-web/issues/3239[#3239] [form] Add missing reference widget view builder
+- https://github.com/eclipse-sirius/sirius-web/issues/3267[#3267] [tree] Fix an issue where `null` values were not filtered in `TreeComponent`.
```
Comment (sbegaudeau) [inline CHANGELOG.adoc:118]: The goal of the representation cannot be to be prevent any potential mistakes that can occur. The contract is quite clear on the tree side "give us a list of elements to consider". So why are there null values in here?

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: CHANGELOG.adoc (hunk 1 above)
Comment (sbegaudeau) [reply CHANGELOG.adoc:118]: And if the issue is coming from the related elements view, that's where it should be fixed and if things are working nicely in the related elements view (for example, we ask for something in the EMF Edit part and we receive invalid values) then it should be fixed in the metamodel.

Comment (AxelRICHARD) [reply CHANGELOG.adoc:118]: Indeed, I have a derived reference in my metamodel who can return `null` values. I fixed it and everything seems ok now.

**Comment by AxelRICHARD** [comment]
A derived reference in a metamodel who can return null values may lead to this NPE.
The solution is to fix the derived reference.

**Pull request closed without merging**